src/tabset/tabset.ts

```diff
@@ -90,7 +90,10 @@ export interface NgbTabChangeEvent {
     </ul>
     <div class="tab-content">
       <template ngFor let-tab [ngForOf]="tabs">
-        <div class="tab-pane active" *ngIf="tab.id === activeId" role="tabpanel"
+        <div class="tab-pane active"
+          *ngIf="preserveNodes || tab.id === activeId"
+          [attr.hidden]="isHidden(tab.id)"
```

Comment:
You should _not_ be binding to the `hidden` attribute but rather use Bootstrap's `active` / `show` CSS classes. Check the markup generated by Bootstrap 4 and follow it: https://v4-alpha.getbootstrap.com/components/navs/#javascript-behavior